src/apps/vpn/controller.cpp

```diff
@@ -219,6 +220,30 @@ bool Controller::activate(const ServerData& serverData,
       m_portalDetected = false;
       return true;
     }
+
+    // Before attempting to enable VPN connection we should check that the
+    // subscription is active.
+    setState(StateCheckSubscription);
+
+    // Set up a network request to check the subscription status.
+    // "task" is an empty task function which is being used to
+    // replicate the behavior of a TaskAccount.
+    TaskFunction* task = new TaskFunction([]() {});
+    NetworkRequest* request = new NetworkRequest(task, 200);
+    request->auth(MozillaVPN::authorizationHeader());
+    request->get(AppConstants::apiUrl(AppConstants::Account));
+
+    connect(request, &NetworkRequest::requestFailed, this,
+            [task, this](QNetworkReply::NetworkError error, const QByteArray&) {
+              logger.error() << "Account request failed" << error;
+              REPORTNETWORKERROR(error, m_errorPropagationPolicy, task->name());
```

Comment:
I'm thinking this should be something like

```
REPORTNETWORKERROR(error, ErrorHandler::DoNotPropagateError, "PreActivationSubscriptionCheck");
```

Reply:
I think we should go back to `StateOff` but let me know if you disagree